soc/arm/atmel_sam/same70/soc.c

```diff
@@ -254,6 +258,17 @@ static int atmel_same70_init(struct device *arg)
 
 	irq_unlock(key);
 
+	/* Check that the CHIP revision matches the HAL revision */
+	bool hal_revb = IS_ENABLED(CONFIG_SOC_ATMEL_SAME70_REVB);
+	bool chip_revb = CHIPID_CIDR_VERSION(CHIPID->CHIPID_CIDR) == 1;
+
+	if (hal_revb != chip_revb) {
```

Comment:
Maybe it would be better to do a full check as in
```
	if (CHIPID->CHIPID_CIDR != CHIP_CIDR) {
```

It makes the code simpler and also covers cases where we try to flash firmware on a device with different flash or RAM size.

Reply:
Thanks for the idea, I just implemented that.